dapp/src/components/dialogs/PurchaseNameDialog.tsx

```diff
@@ -271,12 +285,18 @@ export function PurchaseNameDialog({ name, open, setOpen, onPurchase }: Purchase
                                 </div>
                             </Panel>
                             <div className="flex flex-row gap-x-sm w-full">
-                                <DisplayStats label="Registration Expires" value={expirationDate} />
+                                <DisplayStats
+                                    label="Registration Expires"
+                                    value={<LabelText text={expirationDate} label={`\u00A0`} />} // \u00A0 for alignment
+                                />
                                 <DisplayStats
                                     label="Total Due"
                                     value={
-                                        !isLoadingData && finalPrice > 0 && finalPrice ? (
-                                            formatNanosToIota(finalPrice, { formatRounded: false })
+                                        fiatPriceResult && finalPriceIota ? (
```

Comment:
We should only display the fiat price if it exists. Right now Total Due always falls back to the loading spinner if we’re not on the correct network
```suggestion
          finalPriceIota ? (
                                            <LabelText
                                                text={finalPriceIota}
                                                label={
                                                    fiatPriceResult
                                                        ? `($${fiatPriceResult} USD)`
                                                        : ''
                                                }
                                            />
                                        ) : (
                                            <LoadingIndicator />
                                        )                            
```
<img width="527" height="267" alt="image" src="https://github.com/user-attachments/assets/c172a7ec-d6f4-4996-9c0d-af408e9ce61e" />